### PR TITLE
Add support for multiple rest sensors with a single endpoint

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Add support for multiple rest sensors with a single endpoint

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/46711
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by bdraco** [comment]
I renamed `rest.markdown` to `sensor.rest.markdomain`. It seems to have caused an issue with the rendering but I can't for the life of me figure out why:

```
--- original.markdown	2021-02-17 16:41:52.605941215 -1000
+++ sensor.rest.markdomain	2021-02-17 16:41:57.240974588 -1000
@@ -1,20 +1,17 @@
 ---
-title: RESTful
+title: "RESTful Sensor"
 description: Instructions on how to integrate REST sensors into Home Assistant.
 ha_category:
   - Sensor
 ha_release: 0.7.4
 ha_iot_class: Local Polling
 ha_domain: rest
-ha_platforms:
-  - binary_sensor
-  - notify
-  - sensor
-  - switch
 ---
 
 The `rest` sensor platform is consuming a given endpoint which is exposed by a [RESTful API](https://en.wikipedia.org/wiki/Representational_state_transfer) of a device, an application, or a web service. The sensor has support for GET and POST requests.
 
+_Tip:_ If you want to create multiple `sensors` using the same endpoint, use the [RESTful](/integrations/rest) configuration instructions.
+
 To enable this sensor, add the following lines to your `configuration.yaml` file for a GET request:
 
 ```yaml
```

File: source/_integrations/rest.markdown [hunk 1]
```diff
@@ -1,10 +1,14 @@
 ---
 title: RESTful
-description: Instructions on how to integrate REST sensors into Home Assistant.
+description: Instructions on how to setup rest sensors within Home Assistant.
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
description: Instructions on how to set up rest sensors within Home Assistant.
```

File: source/_integrations/rest.markdown [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 ha_category:
   - Sensor
-ha_release: 0.7.4
+  - Binary Sensor
+ha_release: 2021.3
```
Comment (frenck) [inline source/_integrations/rest.markdown]: This doesn't change :)
```suggestion
ha_release: 0.7.4
```

File: source/_integrations/rest.markdown [hunk 3]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
 
 The `rest` sensor platform is consuming a given endpoint which is exposed by a [RESTful API](https://en.wikipedia.org/wiki/Representational_state_transfer) of a device, an application, or a web service. The sensor has support for GET and POST requests.
 
-To enable this sensor, add the following lines to your `configuration.yaml` file for a GET request:
+[RESTful Sensor](/integrations/sensor.rest) and [RESTful Binary_sensor](/integrations/binary_sensor.rest) can also be setup as platforms if there is only a single sensor per endpoint.
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
[RESTful Sensor](/integrations/sensor.rest) and [RESTful Binary_sensor](/integrations/binary_sensor.rest) can also be set up as platforms if there is only a single sensor per endpoint.
```

File: source/_integrations/sensor.rest.markdomain [hunk 4]
```diff
@@ -0,0 +1,513 @@
+---
```
Comment (frenck) [inline source/_integrations/sensor.rest.markdomain]: We are trying to get away from platform-specific pages. If anything, all other platforms should be merged into `rest.markdown`.

File: source/_integrations/rest.markdown [hunk 5]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
 # Example configuration.yaml entry
-sensor:
-  - platform: rest
-    resource: http://IP_ADDRESS/ENDPOINT
+rest:
+  - authentication: basic
+    username: admin
+    password: password
+    scan_interval: 60
+    resource: http://192.168.1.12/status.xml
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
  - authentication: basic
    username: "admin"
    password: "password"
    scan_interval: 60
    resource: http://192.168.1.12/status.xml
```

File: source/_integrations/rest.markdown [hunk 6]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+    sensor:
+      - name: Adult Pool Data System
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
      - name: "Adult Pool Data System"
```

File: source/_integrations/rest.markdown [hunk 7]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+        json_attributes_path: "$.response.system"
+        value_template: 'OK'
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
        value_template: "OK"
```

File: source/_integrations/rest.markdown [hunk 8]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+          - "time"
+          - "sensor1"
+          - "sensor2"
+          - "sensor3"
+          - "sensor4"
+          - "sensor5"
+          - "version"
+      - name: Adult Pool Data Equipment
+        json_attributes_path: "$.response.equipment"
+        value_template: 'OK'
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
      - name: "Adult Pool Data Equipment"
        json_attributes_path: "$.response.equipment"
        value_template: "OK"
```

File: source/_integrations/rest.markdown [hunk 9]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+          - "circuit2"
+          - "circuit3"
+          - "circuit4"
+          - "circuit5"
+          - "circuit6"
+          - "circuit7"
+          - "circuit8"
+      - name: Adult Pool Data Temp
+        json_attributes_path: "$.response.temp"
+        value_template: 'OK'
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
      - name: "Adult Pool Data Temp"
        json_attributes_path: "$.response.temp"
        value_template: "OK"
```

File: source/_integrations/rest.markdown [hunk 10]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+          - "poolsp"
+          - "spasp"
+          - "pooltemp"
+          - "spatemp"
+          - "airtemp"
+  - authentication: basic
+    username: admin
+    password: password
+    scan_interval: 60
+    resource: http://192.168.1.13/status.xml
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
    username: "admin"
    password: "password"
    scan_interval: 60
    resource: "http://192.168.1.13/status.xml"
```

File: source/_integrations/rest.markdown [hunk 11]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+    sensor:
+      - name: Kiddie Pool Data System
+        json_attributes_path: "$.response.system"
+        value_template: 'OK'
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
      - name: "Kiddie Pool Data System"
        json_attributes_path: "$.response.system"
        value_template: "OK"
```

File: source/_integrations/rest.markdown [hunk 12]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+          - "freeze"
+          - "time"
+          - "sensor1"
+          - "sensor2"
+          - "sensor3"
+          - "sensor4"
+          - "version"
+      - name: Kiddie Pool Data Equipment
+        json_attributes_path: "$.response.equipment"
+        value_template: 'OK'
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
      - name: "Kiddie Pool Data Equipment"
        json_attributes_path: "$.response.equipment"
        value_template: "OK"
```

File: source/_integrations/rest.markdown [hunk 13]
```diff
@@ -15,39 +19,98 @@ ha_platforms:
...
+          - "circuit2"
+          - "circuit3"
+          - "circuit4"
+          - "circuit5"
+          - "circuit6"
+          - "circuit7"
+          - "circuit8"
+      - name: Kiddie Pool Data Temp
+        json_attributes_path: "$.response.temp"
+        value_template: 'OK'
```
Comment (frenck) [inline source/_integrations/rest.markdown]:
```suggestion
      - name: "Kiddie Pool Data Temp"
        json_attributes_path: "$.response.temp"
        value_template: "OK"
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/sensor.rest.markdomain (hunk 4 above)
Comment (bdraco) [reply source/_integrations/sensor.rest.markdomain]: That makes sense. If they get to one of the pages they might not discover both options.  Do you want me to merge them in this PR or in a followup?

**Comment by bdraco** [comment]
hmm

The deploy still looks mangled

https://deploy-preview-16632--home-assistant-docs.netlify.app/integrations/sensor.rest/

I'm at a loss as to whats amiss.  I'll have to come back to this as my work day is getting busy

File: source/_integrations/sensor.rest.markdomain (hunk 4 above)
Comment (frenck) [reply source/_integrations/sensor.rest.markdomain]: I don't know honestly. We don't have to merge all pages right now, but we should not start creating new (old) platform pages.

File: source/_integrations/sensor.rest.markdomain [hunk 14]
```diff
@@ -0,0 +1,513 @@
...
+  - platform: template
+    switches:
+      steam:
+        value_template: '{{ states.sensor.steam_system_data.attributes["usr0"] | int >= 1 }}'
+        turn_on:
+          - service: rest_command.set_steam_led
+            data:
+               led: 6
+          - service: homeassistant.update_entity
+            data:
```
Comment (bdraco) [inline source/_integrations/sensor.rest.markdomain]:
```suggestion
            target:
```

File: source/_integrations/sensor.rest.markdomain [hunk 15]
```diff
@@ -0,0 +1,513 @@
...
+               entity_id: sensor.steam_system_data
+          - delay: 00:00:15
+          - service: homeassistant.update_entity
+            data:
```
Comment (bdraco) [inline source/_integrations/sensor.rest.markdomain]:
```suggestion
            target:
```

File: source/_integrations/sensor.rest.markdomain [hunk 16]
```diff
@@ -0,0 +1,513 @@
...
+               entity_id: sensor.steam_system_data
+        turn_off:
+          - service: rest_command.set_steam_led
+            data:
+               led: 7
+          - service: homeassistant.update_entity
+            data:
```
Comment (bdraco) [inline source/_integrations/sensor.rest.markdomain]:
```suggestion
            target:
```

File: source/_integrations/sensor.rest.markdomain [hunk 17]
```diff
@@ -0,0 +1,513 @@
...
+               entity_id: sensor.steam_system_data
+          - delay: 00:00:15
+          - service: homeassistant.update_entity
+            data:
```
Comment (bdraco) [inline source/_integrations/sensor.rest.markdomain]:
```suggestion
            target:
```

**Comment by bdraco** [comment]
Screen Shot 2021-02-23 at 9 58 53 PM

After I fixed my braindead file naming mistake, I fixed the name conflict so at least we can tell them apart now.

File: source/_integrations/rest.markdown [hunk 18]
```diff
@@ -112,405 +158,13 @@ headers:
...
   required: false
-  type: [string, list]  
-json_attributes:
-  description: A list of keys to extract values from a JSON dictionary result and then set as sensor attributes. If the endpoint returns XML with the "text/xml", "application/xml" or "application/xhtml+xml" content type, it will automatically be converted to JSON according to this [specification](https://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html)
-  required: false
   type: [string, list]
-json_attributes_path:
-  description: A [JSONPath](https://goessner.net/articles/JsonPath/) that references the location of the `json_attributes` in the JSON content.
+sensor:
+  description: A list of [RESTful Sensor](/integrations/sensor.rest) to create from the shared data. All configuration settings that the supported by `RESTful Sensor` not listed above can be used here.
```
Comment (bdraco) [inline source/_integrations/rest.markdown]:
```suggestion
  description: A list of [RESTful Sensor](/integrations/sensor.rest) to create from the shared data. All configuration settings that the supported by [RESTful Sensor](/integrations/sensor.rest#configuration-variables) not listed above can be used here.
```

File: source/_integrations/rest.markdown [hunk 19]
```diff
@@ -112,405 +158,13 @@ headers:
...
   required: false
-  type: string
-force_update:
-  description: Sends update events even if the value hasn't changed. Useful if you want to have meaningful value graphs in history.
+  type: list
+binary_sensor:
+  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.
```
Comment (bdraco) [inline source/_integrations/rest.markdown:167]:
```suggestion
  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest#configuration-variables) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.
```

Comment (bdraco) [inline source/_integrations/rest.markdown]:
```suggestion
  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by [RESTful Binary_sensor](/integrations/binary_sensor.rest#configuration-variables) not listed above can be used here.
```

**Comment by frenck** [comment]
> After I fixed my braindead file naming mistake, I fixed the name conflict so at least we can tell them apart now.

That won't work, as the titles are synced from the codebase (originate from integration manifests).
I suggest merging this, and I will look into merging the pages before the end of beta.

**Review by frenck — APPROVED** [review]

**Pull request merged**